packages/lumx-react/src/components/notification-block/NotificationBlock.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ReactElement, Ref } from 'react';
+
+import classNames from 'classnames';
+
+import { Theme } from '@lumx/react';
+
+import { COMPONENT_PREFIX } from '@lumx/react/constants';
+
+import { GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
+
+/**
+ * Defines the props of the component.
+ */
+interface NotificationBlockProps extends GenericProps {
+    /** After content element */
+    after?: ReactElement;
+    /** Before content element */
+    before?: ReactElement;
+    /** Date when the notification occurs */
+    date?: string;
+    /** Description message of the notification */
+    description?: string;
+    /** Title of the notification */
+    title?: string;
+    /** Reference passed to the wrapper. */
+    notificationBlockRef?: Ref<HTMLDivElement>;
+}
+
+/**
+ * Define the types of the default props.
+ */
+interface DefaultPropsType extends Partial<NotificationBlockProps> {}
+
+/**
+ * The display name of the component.
+ */
+const COMPONENT_NAME = `${COMPONENT_PREFIX}NotificationBlock`;
+
+/**
+ * The default class name and classes prefix for this component.
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
+
+/**
+ * The default value of props.
+ */
+const DEFAULT_PROPS: DefaultPropsType = {
+    theme: Theme.light,
+};
+
+/**
+ * Render a notification in a block.
+ *
+ * @return The component.
+ */
+const NotificationBlock: React.FC<NotificationBlockProps> = ({
+    after,
+    before,
+    date,
+    description,
+    title,
+    className,
+    notificationBlockRef,
+    theme = DEFAULT_PROPS.theme,
+    ...forwardedProps
+}) => {
+    return (
+        <div
+            {...forwardedProps}
+            ref={notificationBlockRef}
+            className={classNames(
+                className,
+                handleBasicClasses({
+                    prefix: CLASSNAME,
+                    theme,
+                }),
+            )}
+        >
+            {before && <div className={`${CLASSNAME}__before`}>{before}</div>}
+            <div className={`${CLASSNAME}__content`}>
+                <span className={`${CLASSNAME}__title`}>{title}</span>
+                <p className={`${CLASSNAME}__description`}>{description}</p>
+                <span className={`${CLASSNAME}__date`}>{date}</span>
```

Comment:
@matmkian we have an accessibility warning:`Elements must have sufficient color contrast` when the text color is `dark L2` on the lumapps theme
Should we update the color?

Moreover the component style do not handle border cases such as long text (maybe ellipsis) etc... 
Is it something we should take care here?

And just to be tedious, should we rename the `date` prop? Since we do not format it or use it as a date. So maybe the prop can be more generic like `metadata`?

Reply:
For long text, we display a multiline paragraph, I'm not sure it's useful to set a max length and an ellipsis. Maybe for the title as we use a `span` tag.

For the date I use a string (like in Angular), but I don't know if we have something in Lumapps to compute the time spent between the notif update and now (To display: `X mins|hours ago`)

Reply:
I guess we will use the moment function `fromNow` to display what we want. But It's not handled by this component itself

Reply:
I reproduced the Angular behavior by using a String, it brings a bit more flexibility on what is rendered on the element

Reply:
No update on dark L2 as it's a Material Design color variant -> the contrast checker of chrome is a little bit too hard.
I don't see any cases where we should have ellipsis as we support multiline for every props.
Date seems appropriate as we should display a date but I see your point, I let you decide the best solution.

Reply:
It was an example, but if the responsive is good as is, it's good for me too 😃 
I think we will change the prop name, I let @gregorycat find out what fit the best

Thanks for the answer